template.yaml

```diff
@@ -1709,6 +1712,26 @@ Resources:
       FunctionName: !Sub '${PreTokenGenerationHandler.Arn}'
       Principal: cognito-idp.amazonaws.com
 
+  PreTokenGenerationFailedAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: Too many pre token generation requests failed
+      AlarmDescription: If this alarm is triggered, then check cloudwatch logs for pre token generation lambda.
+      AlarmActions:
+        - !Ref SlackSnsArn
+      MetricName: Errors
+      Namespace: AWS/Lambda
+      Statistic: Sum
+      Dimensions:
+        - Name: FunctionName
+          Value: !Ref PreTokenGenerationHandler
+      Period: 60
+      EvaluationPeriods: 5
+      DatapointsToAlarm: 1
+      Threshold: 0
+      ComparisonOperator: GreaterThanThreshold
```

Comment:
Sampler data over 5 minutter. Hvis ett av minuttene har hatt mer enn én failed pålogging så sendes det ut en alarm. 

AWS vil da være i "in alarm" i 5 minutter hvis det da etterpå ikke er flere som tryner.

Sto ingenting i tasken om threshold og perioder etc, så tok noe eksempler fra AWS, men tenker det vil også vise seg om vi må justere etterhvert.